src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -121,12 +128,44 @@ def init_from_file(self) -> None:
 
         self.profiles = profile_jsonc.get("profiles", {})
         self.defaults = profile_jsonc.get("defaults", {})
+        self.schema_property = profile_jsonc.get("$schema", None)
 
         # loading secure props is done in load_profile_properties
         # since we want to try loading secure properties only when
         # we know that the profile has saved properties
         # self.load_secure_props()
 
+    def schema_list(
+        self,
+    ) -> list:
+        """
+        Loads the schema properties
+        in a sorted order according to the priority
+        
+        Returns
+        -------
+        Dictionary
+        
+            Returns the profile properties from schema (prop: value)
+        """
+
+        schema = self.schema_property
+        if schema is None:
+            return []
+
+        with open(schema) as f:
+            schema = json.load(f)
+        profile_props = []
+        schema = dict(schema)
+        
+        for props in schema['properties']['profiles']['patternProperties']["^\\S*$"]["allOf"]:
+            props = props["then"]
+            while "properties" in props:
+                props = props.pop("properties")
+                profile_props = list(props.keys())
+        
+        return profile_props
```

Comment:
The `schema_list` method will return the list of `profile_properties` using the `zowe.schema.json` file